boundary/analyze.py

```diff
@@ -206,7 +206,7 @@ def lookup_if_global(name_and_file):
     func_class['sched_outsider'] = (func_class['mod_fns'] - func_class['insider'] - func_class['border']) | func_class['fn_ptr_optimized']
     func_class['optimized_out'] = func_class['sched_outsider'] - func_class['in_vmlinux']
     func_class['public_user'] = func_class['fn'] - func_class['insider'] - func_class['border']
-    func_class['tainted'] = (func_class['border'] | func_class['insider']) & func_class['in_vmlinux']
+    func_class['tainted'] = (func_class['border'] | func_class['insider']) & func_class['in_vmlinux'] - func_class['init']
```

Comment:
Nice catch. But removing func_class['init'] directly from func_class['insider'] is looks better for me. Then we don't need to treat func_class['init'] as a special case.